Analytics-CSharp/Segment/Analytics/Configuration.cs

```diff
@@ -75,7 +81,19 @@ public class Configuration
             this.apiHost = apiHost;
             this.cdnHost = cdnHost;
             this.exceptionHandler = exceptionHandler;
-            this.storageProvider = storageProvider ?? new DefaultStorageProvider();
+
+            if (storageProvider != null)
```

Comment:
the more I think the more I feel we probably only need to smart default persist data path. we should always to default to use storage unless the user specifies to use in memory. what do you think?

Reply:
Yeah, I'm in favor.  More consistent behavior, fewer assumptions and surprises.